### PR TITLE
Rework package indexing

### PR DESCRIPTION
**Pull request by sir-sigurd**
# Description

# TODO

- [x] Unit tests
- [x] [Changelog](CHANGELOG.md) entry

### CODE REVIEW

**Review by akarve — APPROVED** [review]

**Review by nl0 — APPROVED** [review]
will this update (overwrite) "latest" record on every push?

**Comment by sir-sigurd** [comment]
> will this update (overwrite) "latest" record on every push?

Nope.

**Comment by nl0** [comment]
> > will this update (overwrite) "latest" record on every push?
> 
> Nope.

but how will it work? create several "latest" records with different hashes?

**Comment by sir-sigurd** [comment]
> but how will it work? create several "latest" records with different hashes?

Yes, we create a record for every pointer object version (not sure why though).

**Review by nl0 — APPROVED** [review]
looks good

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 * [Changed] Make `pkgpush` lambda directly invocable, adjust handling of parameters and errors ([#2776](https://github.com/quiltdata/quilt/pull/2776))
 * [Changed] Push packages via GraphQL ([#2768](https://github.com/quiltdata/quilt/pull/2768))
 * [Changed] Improve rendering performance for multi-slide .pptx ([#2878](https://github.com/quiltdata/quilt/pull/2878))
+* [Changed] Rework package indexing: now package indexes have documents only for current versions of package pointers objects, documents for 'latest' pointers have `package_hash`, `package_stats`, `comment`, `metadata` fields properly populated ([#2897](https://github.com/quiltdata/quilt/pull/2897))
```
Comment (akarve) [inline docs/CHANGELOG.md:25]: Reading this it's not clear how, e.g., the revision history works in the catalog (if at all). Where are past revisions?

**Review by akarve — APPROVED** [review]

**Pull request merged**